flax/cursor.py

```diff
@@ -327,33 +356,369 @@ def update_fn(path, value):
+  ) -> Generator['Cursor[A]', None, None]:
+    """Traverse the Cursor object and return a generator of child Cursor objects that fulfill the
+    conditions in the ``cond_fn``. The ``cond_fn`` has a function signature of ``(str, Any) -> bool``:
+
+    - The input arguments are the current key path (in the form of a string delimited
+      by ``'/'``) and value at that current key path
+    - The output is a boolean, denoting whether to return the child Cursor object at this path
+
+    NOTES:
+
+    - If the ``cond_fn`` evaluates to True at a particular key path, this method will not recurse
+      any further down that branch; i.e. this method will find and return the "earliest" child nodes
+      that fulfill the condition in ``cond_fn`` in a particular key path
+    - ``.find_all`` WILL NOT search the the value at the top-most level of the pytree (i.e. the root
+      node). The ``cond_fn`` will be evaluated recursively, starting at the root node's children.
+
+    Example::
+
+      import flax.linen as nn
+      from flax.cursor import cursor
+      import jax
+      import jax.numpy as jnp
+
+      class Model(nn.Module):
+        @nn.compact
+        def __call__(self, x):
+          x = nn.Dense(3)(x)
+          x = nn.relu(x)
+          x = nn.Dense(3)(x)
+          x = nn.relu(x)
+          x = nn.Dense(3)(x)
+          x = nn.relu(x)
+          return x
+
+      params = Model().init(jax.random.PRNGKey(0), jnp.empty((1, 2)))['params']
+
+      def cond_fn(path, value):
+        '''Find all dense layer params.'''
+        return 'Dense' in path
+
+      c = cursor(params)
+      for dense_params in c.find_all(cond_fn):
+        dense_params['bias'] += 1
+      new_params = c.build()
+
+      for layer in ('Dense_0', 'Dense_1', 'Dense_2'):
+        assert (new_params[layer]['bias'] == params[layer]['bias'] + 1).all()
+
+      assert jax.tree_util.tree_all(
+            jax.tree_util.tree_map(
+                lambda x, y: (x == y).all(),
+                params,
+                Model().init(jax.random.PRNGKey(0), jnp.empty((1, 2)))[
+                    'params'
+                ],
+            )
+        ) # make sure original params are unchanged
+
+    Args:
+      cond_fn: the function that will conditionally find child Cursor objects
+    Returns:
+      A generator of child Cursor objects that fulfill the condition in the ``cond_fn``.
+    """
+    for path in _traverse_tree((), self._obj, cond_fn=cond_fn):
+      child = self
+      for key, access_type in path:
+        if access_type is AccessType.ITEM:
+          child = child[key]
+        else:  # access_type is AccessType.ATTR
+          child = getattr(child, key)
+      yield child
+
+  def __str__(self):
+    return str(self._obj)
+
+  def __repr__(self):
+    return self.pretty_repr()
+
+  def pretty_repr(self, indent=0):
+    obj_str = (
+        "'" + self._obj + "'" if isinstance(self._obj, str) else str(self._obj)
+    )
```

Comment:
Maybe this could be simplified to:
```suggestion
    obj_str = repr(self._obj)
```
However, if `obj_str` includes new-line characters, these wont respect the indent. You can fix by calling something like:

```python
obj_str = obj_str.replace('\n', '\n' + ' ' * indent)
```